etna/lib/etna/client.rb

```diff
@@ -96,7 +96,12 @@ def body_request(type, endpoint, params = {}, &block)
 
     def query_request(type, endpoint, params = {}, &block)
       uri = request_uri(endpoint)
-      uri.query = URI.encode_www_form(params)
+
+      if uri.query
```

Comment:
Necessary as part of adding upload and download to metis clients in the etna gem.  Because the upload url alraedy encodes hmac params, this logic ensures that any query that is part of the endpoint url is appended to, not removed, by other given params.